src/main/java/io/thinkit/edc/client/connector/services/EdrCache.java

```diff
@@ -0,0 +1,106 @@
+package io.thinkit.edc.client.connector.services;
+
+import static io.thinkit.edc.client.connector.utils.JsonLdUtil.compact;
+import static java.net.http.HttpRequest.BodyPublishers.ofString;
+
+import com.apicatalog.jsonld.JsonLdError;
+import io.thinkit.edc.client.connector.model.*;
+import io.thinkit.edc.client.connector.utils.JsonLdUtil;
+import jakarta.json.JsonArray;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.UnaryOperator;
+
+public class EdrCache {
+    private final String url;
+    private final ManagementApiHttpClient managementApiHttpClient;
+
+    public EdrCache(String url, HttpClient httpClient, UnaryOperator<HttpRequest.Builder> interceptor) {
+        managementApiHttpClient = new ManagementApiHttpClient(httpClient, interceptor);
+        this.url = url;
+    }
+
+    public Result<String> delete(String transferProcessId) {
+        var requestBuilder = HttpRequest.newBuilder()
+                .uri(URI.create("%s/v1/edrs/%s".formatted(this.url, transferProcessId)))
+                .DELETE();
+
+        return this.managementApiHttpClient.send(requestBuilder).map(result -> transferProcessId);
+    }
+
+    public CompletableFuture<Result<String>> deleteAsync(String transferProcessId) {
+        var requestBuilder = HttpRequest.newBuilder()
+                .uri(URI.create("%s/v1/edrs/%s".formatted(this.url, transferProcessId)))
+                .DELETE();
+
+        return this.managementApiHttpClient
+                .sendAsync(requestBuilder)
+                .thenApply(result -> result.map(content -> transferProcessId));
+    }
+
+    public Result<DataAddress> dataAddress(String transferProcessId) {
+        var requestBuilder = HttpRequest.newBuilder()
+                .uri(URI.create("%s/v1/edrs/%s/dataaddress".formatted(this.url, transferProcessId)))
+                .GET();
+        return this.managementApiHttpClient
+                .send(requestBuilder)
+                .map(JsonLdUtil::expand)
+                .map(this::getDataAddress);
+    }
+
+    public CompletableFuture<Result<DataAddress>> dataAddressAsync(String transferProcessId) {
+        var requestBuilder = HttpRequest.newBuilder()
+                .uri(URI.create("%s/v1/edrs/%s/dataaddress".formatted(this.url, transferProcessId)))
+                .GET();
+
+        return this.managementApiHttpClient.sendAsync(requestBuilder).thenApply(result -> result.map(JsonLdUtil::expand)
+                .map(this::getDataAddress));
+    }
+
+    public Result<Edr> request(QuerySpec input) {
+        try {
```

Comment:
why do these two methods need to catch an exception? If it is because of the `compact` method, maybe it's better to catch it there and just throw a runtime exception so callers won't need to do that by themselves